Source/WebKit/Shared/Cocoa/AccessibilityPreferences.mm

```diff
@@ -0,0 +1,116 @@
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "AccessibilityPreferences.h"
+
+#import "AccessibilitySupportSPI.h"
+#import <wtf/SoftLinking.h>
+
+#if ENABLE(ACCESSIBILITY_ANIMATION_CONTROL)
+SOFT_LINK_LIBRARY_OPTIONAL(libAccessibility)
+SOFT_LINK_OPTIONAL(libAccessibility, _AXSReduceMotionAutoplayAnimatedImagesEnabled, Boolean, (), ());
+#endif
+
+#import <pal/spi/cocoa/AccessibilitySupportSoftLink.h>
+
+namespace AXPreferenceHelpers {
+
+static bool shouldUseDefault()
+{
+    NSNumber *forceDefault = [[NSUserDefaults standardUserDefaults] objectForKey:@"ForceDefaultAccessibilitySettings"];
+    return forceDefault ? [forceDefault boolValue] : false;
+}
+
+#if HAVE(PER_APP_ACCESSIBILITY_PREFERENCES)
+WebKitAXValueState reduceMotionEnabled()
+{
+    if (shouldUseDefault()) [[unlikely]]
+        return WebKit::defaultPerAppSettingsState;
+    auto appId = applicationBundleIdentifier().createCFString();
+    return toWebKitAXValueState(_AXSReduceMotionEnabledApp(appId.get()));
+}
+
+WebKitAXValueState increaseButtonLegibility()
+{
+    if (shouldUseDefault()) [[unlikely]]
+        return WebKit::defaultPerAppSettingsState;
+    auto appId = applicationBundleIdentifier().createCFString();
+    return toWebKitAXValueState(_AXSIncreaseButtonLegibilityApp(appId.get()));
+}
+
+WebKitAXValueState enhanceTextLegibility()
+{
+    if (shouldUseDefault()) [[unlikely]]
+        return WebKit::defaultPerAppSettingsState;
+    auto appId = applicationBundleIdentifier().createCFString();
+    return toWebKitAXValueState(_AXSEnhanceTextLegibilityEnabledApp(appId.get()));
+}
+
+WebKitAXValueState darkenSystemColors()
+{
+    if (shouldUseDefault()) [[unlikely]]
+        return WebKit::defaultPerAppSettingsState;
+    auto appId = applicationBundleIdentifier().createCFString();
+    return toWebKitAXValueState(_AXDarkenSystemColorsApp(appId.get()));
+}
+
+WebKitAXValueState invertColorsEnabled()
+{
+    if (shouldUseDefault()) [[unlikely]]
+        return WebKit::defaultPerAppSettingsState;
+    auto appId = applicationBundleIdentifier().createCFString();
+    return toWebKitAXValueState(_AXSInvertColorsEnabledApp(appId.get()));
+}
+#endif // HAVE(PER_APP_ACCESSIBILITY_PREFERENCES)
+
+bool imageAnimationEnabled()
+{
+#if ENABLE(ACCESSIBILITY_ANIMATION_CONTROL)
+    if (shouldUseDefault()) [[unlikely]]
+        return WebKit::defaultImageAnimationEnabled;
+    if (auto* functionPointer = _AXSReduceMotionAutoplayAnimatedImagesEnabledPtr())
```

Comment:
do we still need to use this Ptr method?

Reply:
Maybe not, but not 100% sure considering WebKit is backported some releases. Let's change this in a later PR since I just copy-pasted this from its old location.